runtime/executor/src/main/java/org/apache/nemo/runtime/executor/datatransfer/BlockInputReader.java

```diff
@@ -67,14 +67,15 @@
 
   @Override
   public List<CompletableFuture<DataUtil.IteratorWithNumBytes>> read() {
-    final Optional<CommunicationPatternProperty.Value> comValue =
+    final Optional<CommunicationPatternProperty.Value> comValueOptional =
       runtimeEdge.getPropertyValue(CommunicationPatternProperty.class);
+    final CommunicationPatternProperty.Value comValue = comValueOptional.orElseThrow(IllegalStateException::new);
 
-    if (comValue.get().equals(CommunicationPatternProperty.Value.ONE_TO_ONE)) {
+    if (comValue.equals(CommunicationPatternProperty.Value.ONE_TO_ONE)) {
```

Comment:
I think it would make more sense to rewrite these lines as `CommunicationPatternProperty.Value.ONE_TO_ONE.equals(comValue)`, as it would prevent null pointer exceptions from happening even if the value of `comValue` is null. Same applies to the lines below.

Reply:
FYI, `comValue`  can't be null.
But it would be good to change it for the style consistency.